torch/csrc/api/include/torch/data/dataloader.h

```diff
@@ -59,6 +60,37 @@ class DataLoader {
     }
   }
 
+  /// Constructs a new `DataLoader` from a `dataset`, `options`, and `Sampler`.
+  /// This constructor must be used with a chunk data set. The `Sampler` is
+  /// default to `BatchSizeSampler` for the dataloader, please check
+  /// `make_chunk_data_loader` below. However, a regular sampler and a chunk
+  /// sampler need to be provided to the `dataset` for it to function properly.
+  /// The data loader simply retrives batches using the batch size.
+  DataLoader(
+      Dataset dataset,
```

Comment:
If this DataLoader class only works with ChunkDataSet, does it make sense to change this constructor input parameter to be an explicit ChunkDataSet. So mis-use of this class can be flagged during compilation.

Reply:
We need to support map operation, which is defined in the Dataset class. The design here is that Dataloader does not care about how the chunks are loaded or any additional APIs needed. All these are Dataset specific constructs. One option I did not implement is that at the constructor, we can type check the Dataset and throw an exception.